weeve-agent-installer.sh

```diff
@@ -14,76 +14,115 @@ trap cleanup EXIT
 # if in case the user have deleted the weeve-agent.service and did not reload the systemd daemon
 sudo systemctl daemon-reload
 
+CURRENT_DIRECTORY=$(pwd)
+WEEVE_AGENT_DIRECTORY="$CURRENT_DIRECTORY"/weeve-agent
+
+SERVICE_FILE=/lib/systemd/system/weeve-agent.service
+
+ARGUMENTS_FILE=/lib/systemd/system/weeve-agent.argconf
+
+CLEANUP="true"
+
 cleanup() {
-  if [ "$PROCESS_COMPLETE" = false ]; then
+  if [ "$CLEANUP" = "true" ]; then
+
     log cleaning up the contents ...
+
+    if RESULT=$(systemctl is-active weeve-agent 2>&1); then
     sudo systemctl stop weeve-agent
     sudo systemctl daemon-reload
-    sudo rm /lib/systemd/system/weeve-agent.service
-    sudo rm /lib/systemd/system/weeve-agent.argconf
-    rm -r ./weeve-agent
+    log weeve-agent service stopped
+    else
+    log weeve-agent service not running
+    fi
+
+    if [ -f "$SERVICE_FILE" ]; then
+    sudo rm "$SERVICE_FILE"
+    log "$SERVICE_FILE" removed
+    else
+    log "$SERVICE_FILE" doesnt exists
+    fi
+
+    if [ -f "$ARGUMENTS_FILE" ]; then
+    sudo rm "$ARGUMENTS_FILE"
+    log "$ARGUMENTS_FILE" removed
+    else
+    log "$ARGUMENTS_FILE" doesnt exists
+    fi
```

Comment:
In entire script move common repeated code to function and call that function are required. It will be easy to maintain code and good coding practice

Reply:
there is only one instance of such repetition in the script, current I will keep this, and will modify based on further changes to come.